src/components/Header/Header.tsx

```diff
@@ -1,12 +1,34 @@
+import { Heading, HStack, IconButton } from '@chakra-ui/react';
 import React, { VFC } from 'react';
 import { Icon } from '../Icon/Icon';
 import cls from './Header.module.css';
 
+interface Props {
+
+  /** Whether logo and sidebar toggle button are shown. */
+  showLogo: boolean;
+
+  /** Show sidebar. */
+  onSiderbarShow: () => unknown;
+}
+
 /**
  * TODO: Implement header funtionality.
  */
-export const Header: VFC = () => (
+export const Header: VFC<Props> = ({ showLogo, onSiderbarShow }) => (
   <header className={cls.header}>
+    {showLogo && (
+      <HStack justifyContent="space-between" alignItems="center">
+        <IconButton
+          onClick={() => onSiderbarShow()}
```

Comment:
Do you really need to pass this as anonymous function? I think that just `onClick={onSidebarShow}` will work